pkgs/_macro_tool/lib/main.dart

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) 2024, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:args/args.dart';
+import 'package:path/path.dart' as p;
+
+import 'analyzer_macro_runner.dart';
+import 'cfe_macro_runner.dart';
+import 'macro_tool.dart';
+
+final argParser = ArgParser()
+  ..addOption('host',
+      defaultsTo: 'analyzer', help: 'The macro host: "analyzer" or "cfe".')
+  ..addOption('workspace', help: 'Path to workspace.')
+  ..addOption('packageConfig', help: 'Path to package config.')
+  ..addOption('script', help: 'Path to script.')
+  ..addOption('benchmark-iterations',
+      defaultsTo: '5', help: 'Benchmark iterations.');
+
+Future<int> main(List<String> arguments) async {
+  final args = argParser.parse(arguments);
+  final commands = args.rest;
+
+  final host = HostOption.forString(args['host'] as String?);
+  final workspace = args['workspace'] as String?;
+  final packageConfig = args['packageConfig'] as String?;
+  final script = args['script'] as String?;
+
+  if (host == null ||
+      workspace == null ||
+      packageConfig == null ||
```

Comment:
We really shouldn't need this `packageConfig` argument, can we default it to `Isolate.packageConfig`?